### PR TITLE
[Polkadot] Sidecar implementation

### PR DESCRIPTION
**Pull request by henri-ly**

### CODE REVIEW

File: src/api/polkadot/sidecar.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    specVersion: data.specVersion,
+    metadataRpc: data.metadata,
+  };
+};
+
+export const paymentInfo = async (extrinsic: string) => {
+  const url = `${getSidecarUrl()}/transaction/fee-estimate`;
+  const { data } = await network({
+    method: "POST",
+    url,
```
Comment (haammar-ledger) [inline src/api/polkadot/sidecar.js:55]: Shouldn't it be `url: url` ?

File: src/api/polkadot/sidecar.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+ * WIP - Returns all the balances for an account
+ *
+ * @param {*} addr - the account address
+ */
+const getBalances = async (addr: string) => {
+  const balanceInfo = await fetchBalanceInfo(addr);
+
+  return {
+    balance: BigNumber(balanceInfo.free),
+    spendableBalance: BigNumber(balanceInfo.free), // TODO: Need to find info on sidecar
```
Comment (haammar-ledger) [inline src/api/polkadot/sidecar.js:126]: We can assume minimum balance = minimum bonded = 1 DOT.

The computation should be something like:
`spendable = bonded >= MINIMUM_BOND_AMOUNT ? free - fees : free - EXISTENTIAL_DEPOSIT - fees`
(we need the value for `bonded` though)

File: src/api/polkadot/sidecar.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+export const isNewAccount = async (addr: string): Promise<boolean> => {
+  const { nonce, free } = await fetchBalanceInfo(addr);
+
+  return BigNumber(0).isEqualTo(nonce) && BigNumber(0).isEqualTo(free);
+};
+
+export const isControllerAddress = async (addr: string): Promise<boolean> => {
+  const stakingInfo = await fetchStakingInfo(addr);
+
+  return stakingInfo !== null;
```
Comment (haammar-ledger) [inline src/api/polkadot/sidecar.js:186]: `stakingInfo === null` for a stash?...

File: src/api/polkadot/sidecar.js (hunk 1 above)
Comment (henri-ly) [reply src/api/polkadot/sidecar.js:55]: the variable `url` will be translated as `url: url`

File: src/api/polkadot/sidecar.js (hunk 2 above)
Comment (henri-ly) [reply src/api/polkadot/sidecar.js:126]: the websocket was able to deliver the available balance, but guess we can do the operation on our own

File: src/api/polkadot/sidecar.js (hunk 3 above)
Comment (henri-ly) [reply src/api/polkadot/sidecar.js:186]: Not really, is more like if we got a 404 from this API call, it means the address doesn't have any bond so it does mean neither controller nor stash

File: src/api/polkadot/sidecar.js (hunk 2 above)
Comment (haammar-ledger) [reply src/api/polkadot/sidecar.js:126]: Ok, it just depends if `balanceInfo.free` is the available or the spendable balance (i.e does it include the 1 DOT existential deposit or not)

Comment (henri-ly) [reply src/api/polkadot/sidecar.js:126]: free includes every balance even the bonded one, so yeah let's do the math by hand i guess

**Comment by rjarasson-ledger** [comment]
getStakingProgress to add

**Pull request merged**